nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/ScanKudu.java

```diff
@@ -0,0 +1,376 @@
+
+    protected KuduTable kuduTable;
+
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) throws LoginException {
+        batchSize = context.getProperty(BATCH_SIZE).evaluateAttributeExpressions().asInteger();
+        createKuduClient(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        final KerberosUser user = getKerberosUser();
+        if (user == null) {
+            trigger(context, session);
+            return;
+        }
+
+        final PrivilegedExceptionAction<Void> privelegedAction = () -> {
+            trigger(context, session);
+            return null;
+        };
+
+        final KerberosAction<Void> action = new KerberosAction<>(user, privelegedAction, getLogger());
+        action.execute();
+    }
+
+    public void trigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final List<String> projctedColumnNames = Arrays.asList(context.getProperty(PROJECTED_COLUMNS).evaluateAttributeExpressions(flowFile).getValue().split(","));
+        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        if (StringUtils.isBlank(tableName)) {
+            getLogger().error("Table Name is blank or null for {}, transferring to failure", new Object[] {flowFile});
+            session.transfer(session.penalize(flowFile), REL_FAILURE);
+            return;
+        }
+
+        try {
+            this.kuduTable = getKuduClient().openTable(tableName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        String predicate = context.getProperty(PREDICATES).evaluateAttributeExpressions(flowFile).getValue();
+        final AtomicReference<Long> rowsPulledHolder = new AtomicReference<>(0L);
+        final AtomicReference<Long> ffCountHolder = new AtomicReference<>(0L);
+        ScanKuduResultHandler handler = new ScanKuduResultHandler(session, flowFile, rowsPulledHolder, ffCountHolder, tableName, batchSize);
+
+        try {
+            scan(context,
+                session,
+                this.kuduTable,
+                predicate,
+                projctedColumnNames,
+                handler);
+
+        } catch (Exception e) {
+            if (handler.getFlowFile() != null){
+                session.remove(handler.getFlowFile());
+            }
+            getLogger().error("Unable to fetch rows from Kudu table {} due to {}", new Object[] {tableName, e});
+            flowFile = session.putAttribute(flowFile, "scankudu.results.found", Boolean.toString(handler.isHandledAny()));
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        batchSize = context.getProperty(BATCH_SIZE).evaluateAttributeExpressions(flowFile).asInteger();
```

Comment:
Why do you evaluate this property after L201? Right now, it seems like whatever value is set by the user, the first execution of the processor would use 500 (the default) as the batch size, no?